src/dungeon.cpp

```diff
@@ -1302,36 +1302,39 @@ static void commandQuit() {
     }
 }
 
-static void commandPreviousMessage() {
-    int maxMessages = MESSAGE_HISTORY_SIZE;
-
+static uint8_t calculateMaxMessageCount () {
+    uint8_t max_messages = MESSAGE_HISTORY_SIZE;
     if (command_count > 0) {
-        maxMessages = command_count;
-        if (maxMessages > MESSAGE_HISTORY_SIZE) {
-            maxMessages = MESSAGE_HISTORY_SIZE;
+        if (command_count > MESSAGE_HISTORY_SIZE) {
+            max_messages = MESSAGE_HISTORY_SIZE;
+        } else {
+            max_messages = (uint8_t) command_count;
         }
         command_count = 0;
     } else if (last_command != CTRL_KEY('P')) {
-        maxMessages = 1;
+        max_messages = 1;
     }
+    return max_messages;
+}
```

Comment:
I was having a final look over this and noticed that we could do better on this function. How about this?
```
static uint8_t calculateMaxMessageCount() {
    uint8_t max_messages = MESSAGE_HISTORY_SIZE;

    if (command_count > 0) {
        if (command_count <= MESSAGE_HISTORY_SIZE) {
            max_messages = (uint8_t) command_count;
        }
        command_count = 0;
    } else if (last_command != CTRL_KEY('P')) {
        max_messages = 1;
    }

    return max_messages;
}
```

`max_messages` is assigned `MESSAGE_HISTORY_SIZE` at the top of the function, so we don't actually need to handle that case.